File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/howto/samples/adoplus/adorstodataset/VB/adorstodataset.vb
'-----------------------------------------------------------------------
'  This file is part of the Microsoft .NET SDK Code Samples.
' 
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
'This source code is intended only as a supplement to Microsoft
'Development Tools and/or on-line documentation.  See these other
'materials for detailed information regarding Microsoft code samples.
' 
'THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'-----------------------------------------------------------------------

Imports System
Imports System.Data
Imports System.Data.OleDb
Imports System.Reflection
Imports Microsoft.VisualBasic

namespace HowTo.Samples.ADONET

public class adorstodataset
  public shared sub Main()
    Dim myadorstodataset as adorstodataset
    myadorstodataset = new adorstodataset()
    myadorstodataset.Run()
  end sub

  public sub Run()
    Dim rsType as Type = Type.GetTypeFromProgID("ADODB.RecordSet")
    Dim rsObj as object = Activator.CreateInstance(rsType)

    Dim constr as String = "server=(local)\NetSDK;Integrated Security=SSPI;database=northwind;provider=sqloledb"
    Dim values as Object() = New Object() {"Region", constr, 0, 1, 512}
    rsType.InvokeMember("Open", BindingFlags.InvokeMethod, Nothing, rsObj, values)

    try
        Dim myDataSet as DataSet = new DataSet()
        Dim adapter as OleDbDataAdapter = new OleDbDataAdapter()
        adapter.MissingSchemaAction = MissingSchemaAction.AddWithKey
        Dim count as Integer = adapter.Fill(myDataSet, rsObj, "ADODB.RecordSet")
        Console.WriteLine("FillCount=" & (count))
        Console.WriteLine()
        Console.WriteLine(myDataSet.GetXml())

    catch e as Exception
      Console.Write(e.ToString())

    finally

      try
        'rsType.InvokeMember("Close", BindingFlags.InvokeMethod, Nothing, rsObj, new object(0))
      catch e as Exception
        Console.Write(e.ToString())
      end try
    end try

  end sub
end class

end namespace

File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/howto/samples/adoplus/adorstodataset/CS/adorstodataset.cs
//-----------------------------------------------------------------------
//  This file is part of the Microsoft .NET SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//-----------------------------------------------------------------------

namespace HowTo.Samples.ADONET
{

using System;
using System.Data;
using System.Data.OleDb;
using System.Reflection;

public class adorstodataset
{
  public static void Main()
  {
    adorstodataset myadorstodataset = new adorstodataset();
    myadorstodataset.Run();
  }

  public void Run()
  {
    Type rsType = Type.GetTypeFromProgID("ADODB.RecordSet");
    object rsObj = Activator.CreateInstance(rsType);

    String constr = "server=(local)\\NetSDK;Integrated Security=SSPI;database=Northwind;provider=sqloledb";
    object[] values = new object[] {"Region", constr, /*adOpenForwardOnly*/0, /*adLockReadOnly*/1, 0x200};
    rsType.InvokeMember("Open", BindingFlags.InvokeMethod, null, rsObj, values);

    try
    {
        DataSet myDataSet = new DataSet();
        OleDbDataAdapter adapter = new OleDbDataAdapter();
        adapter.MissingSchemaAction = MissingSchemaAction.AddWithKey;
        int count = adapter.Fill(myDataSet, rsObj, "ADODB.RecordSet");
        Console.WriteLine("FillCount=" + (count));
        Console.WriteLine();
        Console.WriteLine(myDataSet.GetXml());
    }
    catch(Exception e)
    {
      Console.Write(e.ToString());
    }
    finally
    {
      try
      {
          //rsType.InvokeMember("Close", BindingFlags.InvokeMethod, null, rsObj, new object[0]);
      }
      catch(Exception e)
      {
          Console.Write(e.ToString());
      }
    }
  }
}

}
